affinity/cpp-20/d0796r1.md

```diff
@@ -94,6 +102,66 @@ There are some additional challenges which we have been investigating but are no
 * Migrating data from memory allocated in one partition to another
 * Defining memory placement algorithms or policies
 
+
+## Proposed Wording
+
+### Header synopsis
+
+```cpp
+namespace std {
+namespace experimental {
+namespace execution {
+
+/* Execution resource */
+
+struct execution_resource {
+
+  execution_resource() = delete;
+  execution_resource(const execution_resource &) = delete;
+  execution_resource(execution_resource &&) = delete;
+  execution_resource &operator=(const execution_resource &) = delete;
+  execution_resource &operator=(execution_resource &&) = delete;
+
+  size_t concurrency() const noexcept;
+  size_t partition_size() const noexcept;
+
+  const execution_resource &partition(size_t i) const noexcept;
+  const execution_resource &member_of() const noexcept;
+
+  std::string name() const noexcept;
+
+  bool can_place_memory() const noexcept;
+  bool can_place_agent() const noexcept;
+
+};
+
+/* Execution context */
+
+struct execution_context {
+
+  using executor_type = __unspecfied__;
+
+  template <typename ExecutionResource>
+  execution_context(ExecutionResource &&execResource);
```

Comment:
Why not receive the execution resource by value?

Reply:
The execution_resource is not "owned" by the context, only "used", hence by-val semantics make more sense to me